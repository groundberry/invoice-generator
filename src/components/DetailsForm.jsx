import React from "react";
import PropTypes from "prop-types";
import { withStyles } from "@material-ui/core/styles";
import TextField from "@material-ui/core/TextField";

const styles = theme => ({
  textField: {
    display: "flex"
  }
});

class DetailsForm extends React.Component {
  render() {
    const { classes, name, email, street, city, zipCode, phone } = this.props;

    return (
      <form>
        <TextField
          label="Name"
          className={classes.textField}
          name="name"
          value={name}
          onChange={this.handleChange}
          margin="normal"
          variant="outlined"
        />
        <TextField
          label="Email"
          className={classes.textField}
          name="email"
          value={email}
          onChange={this.handleChange}
          type="email"
          autoComplete="email"
          margin="normal"
          variant="outlined"
        />
        <TextField
          label="Street"
          className={classes.textField}
          name="street"
          value={street}
          onChange={this.handleChange}
          margin="normal"
          variant="outlined"
        />
        <TextField
          label="City"
          className={classes.textField}
          name="city"
          value={city}
          onChange={this.handleChange}
          margin="normal"
          variant="outlined"
        />
        <TextField
          label="Zip code"
          className={classes.textField}
          name="zipCode"
          value={zipCode}
          onChange={this.handleChange}
          margin="normal"
          variant="outlined"
        />
        <TextField
          label="Phone"
          className={classes.textField}
          name="phone"
          value={phone}
          onChange={this.handleChange}
          margin="normal"
          variant="outlined"
        />
      </form>
    );
  }

  handleChange = evt => {
    const { name, value } = evt.target;
    this.props.onChange({
      [name]: value
    });
  };
}

DetailsForm.propTypes = {
  classes: PropTypes.object.isRequired,
  name: PropTypes.string,
  email: PropTypes.string,
  street: PropTypes.string,
  city: PropTypes.string,
  zipCode: PropTypes.string,
  phone: PropTypes.string,
  onChange: PropTypes.func
};

export default withStyles(styles)(DetailsForm);
