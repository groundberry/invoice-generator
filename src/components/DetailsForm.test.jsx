import React from "react";
import { createShallow } from "@material-ui/core/test-utils";
import DetailsForm from "./DetailsForm";
import TextField from "@material-ui/core/TextField";

const data = {
  name: "John",
  email: "john@doe.com",
  street: "NE 8th St",
  city: "Bellevue",
  zipCode: "98004",
  phone: "555 666 7777"
};

describe("DetailsForm", () => {
  let wrapper;
  let onChange;

  beforeEach(() => {
    onChange = jest.fn();

    const shallow = createShallow({
      untilSelector: DetailsForm
    });
    wrapper = shallow(
      <DetailsForm
        name={data.name}
        email={data.email}
        street={data.street}
        city={data.city}
        zipCode={data.zipCode}
        phone={data.phone}
        onChange={onChange}
      />
    );
  });

  it("renders", () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe("when name changes", () => {
    beforeEach(() => {
      wrapper
        .find(TextField)
        .filter({ name: "name" })
        .simulate("change", { target: { name: "name", value: "Jane" } });
    });

    it("invokes onChange with the new name", () => {
      expect(onChange).toHaveBeenCalledWith({ name: "Jane" });
    });
  });

  describe("when email changes", () => {
    beforeEach(() => {
      wrapper
        .find(TextField)
        .filter({ name: "email" })
        .simulate("change", { target: { name: "email", value: "jane@doe.com" } });
    });

    it("invokes onChange with the new email", () => {
      expect(onChange).toHaveBeenCalledWith({ email: "jane@doe.com" });
    });
  });

  describe("when street changes", () => {
    beforeEach(() => {
      wrapper
        .find(TextField)
        .filter({ name: "street" })
        .simulate("change", { target: { name: "street", value: "12th Ave" } });
    });

    it("invokes onChange with the new street", () => {
      expect(onChange).toHaveBeenCalledWith({ street: "12th Ave" });
    });
  });

  describe("when city changes", () => {
    beforeEach(() => {
      wrapper
        .find(TextField)
        .filter({ name: "city" })
        .simulate("change", { target: { name: "city", value: "Seattle" } });
    });

    it("invokes onChange with the new city", () => {
      expect(onChange).toHaveBeenCalledWith({ city: "Seattle" });
    });
  });

  describe("when zipCode changes", () => {
    beforeEach(() => {
      wrapper
        .find(TextField)
        .filter({ name: "zipCode" })
        .simulate("change", { target: { name: "zipCode", value: "98122" } });
    });

    it("invokes onChange with the new zipCode", () => {
      expect(onChange).toHaveBeenCalledWith({ zipCode: "98122" });
    });
  });

  describe("when phone changes", () => {
    beforeEach(() => {
      wrapper
        .find(TextField)
        .filter({ name: "phone" })
        .simulate("change", { target: { name: "phone", value: "555 666 8888" } });
    });

    it("invokes onChange with the new phone", () => {
      expect(onChange).toHaveBeenCalledWith({ phone: "555 666 8888" });
    });
  });
});
