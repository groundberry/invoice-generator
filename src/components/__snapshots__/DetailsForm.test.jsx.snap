// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailsForm renders 1`] = `
<form>
  <TextField
    className="DetailsForm-textField-1"
    label="Name"
    margin="normal"
    name="name"
    onChange={[Function]}
    required={false}
    select={false}
    value="John"
    variant="outlined"
  />
  <TextField
    autoComplete="email"
    className="DetailsForm-textField-1"
    label="Email"
    margin="normal"
    name="email"
    onChange={[Function]}
    required={false}
    select={false}
    type="email"
    value="john@doe.com"
    variant="outlined"
  />
  <TextField
    className="DetailsForm-textField-1"
    label="Street"
    margin="normal"
    name="street"
    onChange={[Function]}
    required={false}
    select={false}
    value="NE 8th St"
    variant="outlined"
  />
  <TextField
    className="DetailsForm-textField-1"
    label="City"
    margin="normal"
    name="city"
    onChange={[Function]}
    required={false}
    select={false}
    value="Bellevue"
    variant="outlined"
  />
  <TextField
    className="DetailsForm-textField-1"
    label="Zip code"
    margin="normal"
    name="zipCode"
    onChange={[Function]}
    required={false}
    select={false}
    value="98004"
    variant="outlined"
  />
  <TextField
    className="DetailsForm-textField-1"
    label="Phone"
    margin="normal"
    name="phone"
    onChange={[Function]}
    required={false}
    select={false}
    value="555 666 7777"
    variant="outlined"
  />
</form>
`;
